src/core/Akka/Actor/Futures.cs

```diff
@@ -187,6 +195,72 @@ internal static IActorRefProvider ResolveProvider(ICanTell self)
         }
     }
 
+    /// <summary>
+    /// A very lightweight actor implementation, which sole purpose is to be a conductor
+    /// between an actor world and .net classic world.
+    /// It waits for a single message and sets it to the TaskCompletionSource.
+    /// </summary>
+    internal class AskActorRef : MinimalActorRef
```

Comment:
A simplified copy of FutureActorRef.

I could not use PromiseActorRef since it's doing way too much. The safest way to go here is to be as close to the old implementation of the FutureActorRef.

@Aaronontheweb After implementing this class, I realized that I could just propagate empty unregister action and I would be done. Do you want me to do so?